tests/Action/EntrypointActionTest.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\tests\Action;
+
+use ApiPlatform\Core\Action\EntrypointAction;
+use ApiPlatform\Core\Api\Entrypoint;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceNameCollection;
+
+/**
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ */
+class EntrypointActionTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGetEntrypoint()
+    {
+        $resourceNameCollectionFactory = $this->prophesize(ResourceNameCollectionFactoryInterface::class);
```

Comment:
I would keep the suffix `Prophecy`: it's what's used below and `$resourceNameCollectionFactory` would be a more appropriate name of the stub